frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx

```diff
@@ -31,14 +43,100 @@ const createHelpText = (k8sModel: K8sKind, helpText: string) => {
 const ResourceSection: React.FC<ResourceSectionProps> = ({ flags }) => {
   const { t } = useTranslation();
   const [field] = useField<Resources[]>('resourceTypesNotValid');
+  const fieldName = 'resources';
+  const {
+    values: { formType },
+    setFieldValue,
+  } = useFormikContext<FormikValues>();
   const invalidTypes = field.value || [];
+
   const knativeServiceAccess = useAccessReview({
     group: ServiceModel.apiGroup,
     resource: ServiceModel.plural,
     namespace: getActiveNamespace(),
     verb: 'create',
   });
+  const canIncludeKnative =
+    !invalidTypes.includes(Resources.KnativeService) &&
+    flags[FLAG_KNATIVE_SERVING_SERVICE] &&
+    knativeServiceAccess;
+
+  const defaultResourceType = canIncludeKnative ? Resources.KnativeService : Resources.Kubernetes;
+
+  const [
+    preferredResourceType,
+    setPreferredResourceType,
+    preferredResourceTypeLoaded,
+  ] = useUserSettings<string>(PREFERRED_RESOURCE_TYPE_USER_SETTING_KEY, defaultResourceType, true);
 
+  const [resourceType, setResourceType, resourceTypeLoaded] = useUserSettingsCompatibility<string>(
+    LAST_RESOURCE_TYPE_STORAGE_KEY,
+    PREFERRED_RESOURCE_TYPE_LOCAL_STORAGE_KEY,
+    defaultResourceType,
+    true,
+  );
+
+  React.useEffect(() => {
```

Comment:
do we need this and below two useEffect's and will it make sense to extract this into the custom hook?

Reply:
Yes, we need it. Moved it to a custom hook.